lib/std/fmt.zig

```diff
@@ -1461,6 +1461,88 @@ test "fmtDuration" {
     }
 }
 
+fn formatDurationSigned(ns: i64, comptime fmt: []const u8, options: std.fmt.FormatOptions, writer: anytype) !void {
+    _ = fmt;
+    _ = options;
+    if (ns < 0) {
+        try writer.print("-{}", .{std.fmt.fmtDuration(@intCast(u64, -ns))});
```

Comment:
Emit the Minus sign if needed and then call `formatDuration` to avoid the overhead of the double fmt parsing.

Reply:
Good point, I took this approach of going through `fmtDuration` in my code as `formatDuration` isnt `pub`.